rqt_py_common/src/rqt_py_common/message_field_type_helpers.py

```diff
@@ -0,0 +1,505 @@
+    VALID_BASE_MSG_RE + '|(' + STATIC_ARRAY_RE + ')|(' + \
+    UNBOUNDED_ARRAY_RE + ')|(' + BOUNDED_ARRAY_RE + ')'
+
+COMPILED_STATIC_ARRAY_RE = re.compile(STATIC_ARRAY_RE)
+COMPILED_UNBOUNDED_ARRAY_RE = re.compile(UNBOUNDED_ARRAY_RE)
+COMPILED_BOUNDED_ARRAY_RE = re.compile(BOUNDED_ARRAY_RE)
+COMPILED_VALID_MSG_RE = re.compile(VALID_MSG_RE)
+COMPILED_BOUNDED_STRING_RE = re.compile(BOUNDED_STRING_RE)
+
+class MessageFieldTypeInfo(object):
+    logger = logging.get_logger('MessageFieldTypeInfo')
+
+    __slots__ = (
+        'is_valid',
+        'base_type_str',
+        'is_array',
+        'is_static_array',
+        'static_array_size',
+        'is_bounded_array',
+        'bounded_array_size',
+        'is_unbounded_array',
+        'is_bounded_string',
+        'bounded_string_size',
+    )
+
+    """
+    Useful helper methods for working with a message's field types
+
+    Parse items such as: (from rosidl_generator_py definitions)
+      .msg            field_type                    python
+      -----------------------------------------------------------
+      bool            -> 'boolean'                  -> bool
+      byte            -> 'octet'                    -> bytes
+      float32         -> 'float'                    -> float
+      float64         -> 'double'                   -> float
+      char            -> 'uint8'                    -> int
+      int8            -> 'int8'                     -> int
+      int16           -> 'int16'                    -> int
+      int32           -> 'int32'                    -> int
+      int64           -> 'int64'                    -> int
+      uint8           -> 'uint8'                    -> int
+      uint16          -> 'uint16'                   -> int
+      uint32          -> 'uint32'                   -> int
+      uint64          -> 'uint64'                   -> int
+      string          -> 'string'                   -> str
+      my_msgs/Custom  -> 'my_msgs/Custom'           -> my_msgs.msg.Custom
+
+      # Arrays with normal values
+      int8[3]         -> 'int8[3]'                  -> List[int]
+      int8[]          -> 'sequence<int8>'           -> List[int]
+      int8[<=3]       -> 'sequence<int8, 3>'        -> List[int]
+
+      # String weirdness
+      string<=5       -> 'string<5>'                -> str
+      string[3]       -> 'string[3]'                -> List[str]
+      string[]        -> 'sequence<string>'         -> List[str]
+      string<=5[3]    -> 'string<5>[3]'             -> List[str]
+      string<=5[<=10] -> 'sequence<string<5>, 10>'  -> List[str]
+      string<=5[]     -> 'sequence<string<5>>'      -> List[str]
+      string[<=10]    -> 'sequence<string, 10>'     -> List[str]
+
+    :param field_type: the field type in the form generated by msg.get_fields_and_field_types
+    :type field_type: str
+    """
+    def __init__(self, field_type: str):
+        super(MessageFieldTypeInfo, self).__init__()
+
+        self.is_valid = False
```

Comment:
As member variables, these are publicly accessible and technically writeable. I would make them private and either 1). create public properties with no public setter, or 2) create public getters (i.e. self.is_valid -> self._is_valid, self.is_valid())